yt/yt/server/query_tracker/query_tracker_proxy.h

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+#include "private.h"
+
+#include <yt/yt/client/api/client.h>
+
+#include <yt/yt/ytlib/api/native/public.h>
+
+#include <yt/yt/core/ypath/public.h>
+
+namespace NYT::NQueryTracker {
+
+using namespace NApi;
+using namespace NYPath;
+using namespace NYson;
+
+///////////////////////////////////////////////////////////////////////////////
+
+class TQueryTrackerProxy
+    : public TRefCounted
+{
+public:
+    TQueryTrackerProxy(
+        NNative::IClientPtr stateClient,
+        TYPath stateRoot);
+
+    void StartQuery(
+        const TQueryId queryId,
+        const NQueryTrackerClient::EQueryEngine engine,
+        const TString& query,
+        const TStartQueryOptions& options,
+        const TString& user);
+
+    void AbortQuery(
+        const TQueryId queryId,
+        const TAbortQueryOptions& options,
+        const TString& user);
+
+    TQueryResult GetQueryResult(
+        const TQueryId queryId,
+        const i64 resultIndex,
+        const TString& user);
+
+    IUnversionedRowsetPtr ReadQueryResult(
+        const TQueryId queryId,
+        const i64 resultIndex,
+        const TReadQueryResultOptions& options,
+        const TString& user);
+
+    TQuery GetQuery(
+        const TQueryId queryId,
+        const TGetQueryOptions& options,
+        const TString& user);
+
+    TListQueriesResult ListQueries(
+        const TListQueriesOptions& options,
+        const TString& user);
+
+    void AlterQuery(
+        const TQueryId queryId,
+        const TAlterQueryOptions& options,
+        const TString& user);
+
```

Comment:
At the moment, there is no logic in it that should be assigned to qt (related to the qt state) As soon as it appears, we will add this method

Reply:
I think we should move GetQueryTrackerInfo to QT. The idea is to get info about query tracker version, and this API tries to analyze QT state for it

The logic may also simplify a lot, as we will know QT version from QT code and can hardcode supported features (such as Access Control), instead of invasive QT state analysis (`//sys/ACO/queries` existence check)

For now just copying the current logic is fine. For new features it will be easier to add

Reply:
Added GetQueryTrackerInfo